src/System.Private.ServiceModel/src/System/ServiceModel/Security/CryptoHelper.cs

```diff
@@ -0,0 +1,375 @@
+
+            //}
+            //if (offset < 0 || offset > cipherText.Length - count)
+            //{
+            //    throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new ArgumentOutOfRangeException("offset", SR.GetString(SR.ValueMustBeInRange, 0, cipherText.Length - count)));
+            //}
+
+            //int ivSize = algorithm.BlockSize / 8;
+            //byte[] iv = new byte[ivSize];
+            //Buffer.BlockCopy(cipherText, offset, iv, 0, iv.Length);
+            //algorithm.Padding = PaddingMode.ISO10126;
+            //algorithm.Mode = CipherMode.CBC;
+            //try
+            //{
+            //    using (ICryptoTransform decrTransform = algorithm.CreateDecryptor(algorithm.Key, iv))
+            //    {
+            //        return decrTransform.TransformFinalBlock(cipherText, offset + iv.Length, count - iv.Length);
+            //    }
+            //}
+            //catch (CryptographicException ex)
+            //{
+            //    throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new MessageSecurityException(SR.GetString(SR.DecryptionFailed), ex));
+            //}
+        }
+
+        internal static void FillRandomBytes(byte[] buffer)
+        {
+            throw ExceptionHelper.PlatformNotSupported();   // Issue #31 in progress
+            //random.GetBytes(buffer);
+        }
+
+        static CryptoAlgorithmType GetAlgorithmType(string algorithm)
+        {
+            throw ExceptionHelper.PlatformNotSupported();   // Issue #31 in progress
+
+            //object algorithmObject = null;
+
+            //try
+            //{
+            //    algorithmObject = CryptoAlgorithms.GetAlgorithmFromConfig(algorithm);
+            //}
+            //catch (InvalidOperationException)
+            //{
+            //    algorithmObject = null;
+            //    // We swallow the exception and continue.
+            //}
+            //if (algorithmObject != null)
+            //{
+            //    SymmetricAlgorithm symmetricAlgorithm = algorithmObject as SymmetricAlgorithm;
+            //    KeyedHashAlgorithm keyedHashAlgorithm = algorithmObject as KeyedHashAlgorithm;
+            //    if (symmetricAlgorithm != null || keyedHashAlgorithm != null)
+            //        return CryptoAlgorithmType.Symmetric;
+
+            //    // NOTE: A KeyedHashAlgorithm is symmetric in nature.
+
+            //    AsymmetricAlgorithm asymmetricAlgorithm = algorithmObject as AsymmetricAlgorithm;
+            //    SignatureDescription signatureDescription = algorithmObject as SignatureDescription;
+            //    if (asymmetricAlgorithm != null || signatureDescription != null)
+            //        return CryptoAlgorithmType.Asymmetric;
+
+            //    return CryptoAlgorithmType.Unknown;
+            //}
+
+            //switch (algorithm)
```

Comment:
These algorithms are still all in the latest soap spec and some of them are required to be supported by a soap version which we officially support but you wouldn't want to use them for a real scenario. I believe nothing in that list is deprecated although some is discouraged.

---
In reply to: [78192012](https://github.com/dotnet/wcf/pull/1515#discussion_r78192012) [](ancestors = 78192012,78158972)